SECP-service/src/main/resources/assets/app/js/chat/chat-controller.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+angular.module('SECP')
+  .controller('ChatController', ['$scope', '$modal', 'Chat',
+    function ($scope, $modal, Chat) {
+
+      //declaring variables
+      $scope.contacts = [];
+      $scope.currentUser = Chat.getCurrentUser();
+
+
+      Chat.getChatList().then(function(data) {
+        $scope.contacts = data;
+      });
+
+      $scope.sendMessage = function() {
+         var message = {
+            'text': $scope.messageInput
+         };
+         $scope.messages.slice(-1)[0].contents.push(message)
+         //clearing the message input in the textarea
+         $scope.messageInput = null;
+      };
+
+        $scope.sendMessageUsingEnter = function() {
+            var code = event.keyCode || event.which;
+            if (code === 13) {
+                if (!event.shiftKey) {
+                    event.preventDefault();
+                    $scope.sendMessage();
+                }
+            }
+        };
+
+      $scope.contactSelected = function(contact) {
+         $scope.selectedChat = contact;
+         Chat.getMessages(contact).then(function(data) {
+            $scope.messages = data;
+         });
+      };
+
+      $scope.create = function () {
```

Comment:
we don't need all the methods below contactSelected. I just forgot to delete them.

Reply:
Ill get rid of them